dashboard/test/controllers/api/v1/projects/public_gallery_controller_test.rb

```diff
@@ -0,0 +1,121 @@
+require 'test_helper'
+
+class Api::V1::Projects::PublicGalleryControllerTest < ActionController::TestCase
+  setup do
+    published_applab_project = {
+      id: 33,
+      published_at: '2017-03-03T00:00:00.000-08:00',
+      project_type: 'applab',
+      value: {
+        name: 'Charlies App',
+        thumbnailUrl: 'https://studio.code.org/charlies_thumbnail.png'
+      }.to_json,
+    }
+    stub_projects = [published_applab_project]
+
+    def db_result(result)
+      stub(where: stub(exclude: stub(order: stub(limit: stub(offset: result)))))
+    end
```

Comment:
nit: Avoid [nested methods](http://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Lint/NestedMethodDefinition). Any reason this can't be its own method?